### PR TITLE
Chore: Update arrow and prometheus dependencies

### PR DESCRIPTION
**Pull request by ryantxu**
This PR updates the SDK to a [version with latest arrow files](https://github.com/grafana/grafana-plugin-sdk-go/pull/890).  While doing the import -- it seems to also require upgrades to  `/x/exp` and by extension prometheus 😬 

https://github.com/prometheus/prometheus/issues/12649

Without the prometheus update, the error was:
```
type func(a Label, b Label) bool of func(a, b Label) bool {…} does not match inferred 
type func(a Label, b Label) int for func(a E, b E) int
```

### CODE REVIEW

File: pkg/services/ngalert/sender/notifier.go [hunk 1]
```diff
@@ -336,7 +336,7 @@ func (n *Manager) Send(alerts ...*Alert) {
 			}
 		})
 
-		a.Labels = lb.Labels(a.Labels)
+		a.Labels = lb.Labels()
```
Comment (ryantxu) [inline pkg/services/ngalert/sender/notifier.go:339]: @grafana/alerting-backend-product -- the prometheus upgrade no longer has this function, but also the previous logic did not make sense.  And I don't see any test :(

File: go.mod [hunk 2]
```diff
@@ -63,7 +63,7 @@ require (
 	github.com/grafana/cuetsy v0.1.11 // @grafana/grafana-as-code
 	github.com/grafana/grafana-aws-sdk v0.23.1 // @grafana/aws-datasources
 	github.com/grafana/grafana-azure-sdk-go v1.12.0 // @grafana/partner-datasources
-	github.com/grafana/grafana-plugin-sdk-go v0.208.0 // @grafana/plugins-platform-backend
+	github.com/grafana/grafana-plugin-sdk-go v0.209.1-0.20240208231327-bb7c8753e380 // @grafana/plugins-platform-backend
```
Comment (ryantxu) [inline go.mod:66]: ^^ will be replaced with a real version after the PR is merged

Comment (marefr) [reply go.mod:66]: When the PR is merged you mean?

Comment (itsmylife) [reply go.mod:66]: This PR https://github.com/grafana/grafana-plugin-sdk-go/pull/890

**Review by itsmylife — APPROVED** [review]
influxdb changes LGTM

**Comment by alexweav** [comment]
Alerting bits LGTM, thanks!

File: go.mod (hunk 2 above)
Comment (andresmgot) [reply go.mod:66]: [v0.210.0 released](https://github.com/grafana/grafana-plugin-sdk-go/releases/tag/v0.210.0), you can update this.

**Review by andresmgot — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (itsmylife) [reply go.mod:66]: We will do it on Tuesday

**Pull request merged**